### PR TITLE
Dashboard Page

### LINKED ISSUE #2: Student Dashboard ("SD")
- [x] The SD should have a welcome bar which will display the first name of the signee/student taken during sign up. (let's add this to phase 1 (later iterations).
- [x] The SD should have a hamburger menu for the student which will be enabled once the student goes to any further page from their dashboard. The list of the menu shall be help the student to get back to their dashboard. (not an immediate feature to have: (let's add this to phase 1 (later iterations))
- [x] The SD should have three step options as "Submit your levelZero Portfolio" and "Give your Interview" and "Secure your seat" .
- [x] Below the welcome bar there  should be the "Your status" bar which shall reflect the status of the students application based on the most recent action of the student. 
- [x] The Your Status bar should also have a sub-heading that should state the clear action to be taken by the student as next step.
- [x] The current status path should be : Portfolio not submitted > Portfolio under review > Portfolio needs revision /or Portfolio review completed (Interview step unlocked) > Interview not scheduled > Interview scheduled > Interview results out (Selected for levelOne or Not selected for this batch) : The selected for levelOne (unlock the 3rd step > Admission fee not paid > Admission fee paid /or financial aid application under review > Enrolment completed  (Welcome to levelOne card)
- [x] There should be a "Your to-do" for clarity which would be the clear CTA on what the student has to do to move forward.
- [x] The SD after log-in should have only first step un-locked, i.e. "Submit your portfolio" with current status at the header as " Portfolio not submitted" which shall be updated after the student has submitted their portfolio for review and as per the results of the review and will keep changing based on the status of the student.
- [x] The current status of other two steps should be "locked" and those two steps buttons shall also be disabled. These steps shall only be enabled after their preceding step has been completed, i.e. For step 2 to unlock the status should be "Portfolio review completed and for step 3 to unlock the status should be "Selected for levelOne".
- [x] Prototype of the Welcome Dashboard:
image

The figma component to refer to: https://www.figma.com/file/AvkiFXfsMSEOQFp68p1g6N/ADM-System?node-id=807%3A1757

(NOTE: The prototype will be changed as the current status has to be re-designed.)

- [x] The submit your portfolio button should take the student to the "Submit Portfolio" window.
- [x] Prototype of Submit portfolio window:

image

II. The SUBMIT YOUR PORTFOLIO bar-button:
- [x] The review system of student dashboard shall have only this button enabled.
- [x] Clicking on this button should take the student to the "Submit portfolio" window.
- [x] The "Current Status" as the header below welcome bar shall be the most dynamic part of the student dashboard. For the current status shall be " Portfolio not submitted" > "Portfolio Under Review" > "Portfolio Review Completed: you are ready for interview" (leads to unlock of step2) or "Portfolio Needs Revision: Fix and re-submit" (leads to re-submission window.
- [x] Portfolio Under Review: The Current Status should change to "Portfolio Under Review" only after the student submits their portfolio url for review and the portfolio has been stored in our database. This status shall remain the same unless the reviewer has reviewed the student's portfolio.
- [x] Portfolio Review Completed: The current status should be updated to " Portfolio Review Completed: You are ready for interview" once the Reviewer has reviewed the student's portfolio and clicked on the "mark15" button. This shall also unlock the step 2 on the SD, i.e. "Give your Interview" bar.
- [x] Portfolio Needs Revision: The current status should be updated to " Portfolio Needs Revision : Fix and re-submit" if the Reviewer's chooses the " Needs revision" button after reviewing the student's portfolio.
- [x] If the student clicks on the "Portfolio Needs Revision" status update, it should take the student to the portfolio re-submission window.
- [x] prototype for portfolio submitted and update in current status to "Portfolio under review":
image

- [x] prototype for SD, where portfolio has been reviewed but needs revision and the SD update in current status:
image

- [x] prototype for Re-submission window after the student clicks on that update "Needs Revision":
image

- [x] After the student re-submit the portfolio the current status on the dashboard under step 1 should again change to "Under Review". 
- [x] Prototype of change in current status to "portfolio under review" in SD after re-submission:
image

- [x] After the portfolio review is completed the status shall change to "Portfolio review completed" and the sub-heading would be "Congratulations, the step 2 has been unlocked. Please click on it and give your interview."
- [x] Prototype of the same:
image

Figma file to refer:
https://www.figma.com/file/AvkiFXfsMSEOQFp68p1g6N/ADM-System?node-id=19%3A289
PRD: https://www.notion.so/Product-Spec-Platform-3-4-5f2aaaf2b98b425b89923f9a2b48e649
Wiring : https://www.figma.com/proto/AvkiFXfsMSEOQFp68p1g6N/ADM-System?node-id=80%3A407&scaling=min-zoom&page-id=0%3A1

### PR DESCRIPTION
**Pull request by NadaaFarook**
**What features does this PR Address ?**
Dashboard UI

What issues this PR solves?
Fixes #2 

Where have you tested it?
Edge ( mobile and desktop responsive)

Quick Demo
image

### CODE REVIEW

File: client/src/pages/dashboard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import Image from "next/image";
+import DashboardImage from "../../../images/dashboard-girl.png";
+export default function Dashboard() {
+  const bgColor = "gray.700";
+  const status = {
+    status: "interview_done_passed",
+    level: 1, // control the locks
+    statusText: "You need to submit your portfolio to move ahead", // show your header
+  };
+  const steps = [
```
Comment (uditkumar01) [inline client/src/pages/dashboard/index.tsx:22]: export this data to a separate file and import it from there

File: client/src/styles/Home.module.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
+.container {
```
Comment (uditkumar01) [inline client/src/styles/Home.module.css:1]: Remove this file

File: client/src/styles/globals.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
+html,
```
Comment (uditkumar01) [inline client/src/styles/globals.css:1]: remove this file as well

**Review by uditkumar01 — COMMENTED** [review]
Cool UI. Please make the changes mentioned if possible.

File: client/src/pages/dashboard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      <Flex as="section" flexDir="column" m={8}>
+        <Flex
+          className="welcome-section"
+          bgColor={bgColor}
+          p={8}
+          borderRadius={5}
+          flexDir={["column", "column", "row"]}
+        >
+          <Flex flexDir="column" width={["100%", "100%", "60%"]}>
+            <Heading>Hii Rahul 👋</Heading>
```
Comment (MehulKChaudhari) [inline client/src/pages/dashboard/index.tsx:59]: here name is as per user so `{user.name}` should be there ig

**Review by MehulKChaudhari — COMMENTED** [review]
Good work. 👍

**Pull request closed without merging**